### PR TITLE
ARROW-13018: [C++][Docs] Use consistent terminology for nulls (min_count) in scalar aggregate kernels

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @rok

File: cpp/src/arrow/compute/kernels/aggregate_basic.cc [hunk 1]
```diff
@@ -378,26 +378,25 @@ namespace internal {
...
 
-const FunctionDoc sum_doc{"Sum values of a numeric array",
-                          ("Null values are ignored. Minimum count of non-NA\n"
-                           "values can be set and NAN is returned if too "
-                           "few are present.\n"
-                           "This can be changed through ScalarAggregateOptions."),
-                          {"array"},
-                          "ScalarAggregateOptions"};
+const FunctionDoc sum_doc{
+    "Sum values of a numeric array",
```
Comment (rok) [inline cpp/src/arrow/compute/kernels/aggregate_basic.cc:387]:
```suggestion
    "Compute the sum values of a numeric array",
```

Comment (rok) [reply cpp/src/arrow/compute/kernels/aggregate_basic.cc:387]: it's a bit of a nit for consistency with `Compute the mean`.

**Review by rok — APPROVED** [review]
Looks good!

File: cpp/src/arrow/compute/kernels/aggregate_basic.cc (hunk 1 above)
Comment (jorisvandenbossche) [reply cpp/src/arrow/compute/kernels/aggregate_basic.cc]: Yep, thanks!

**Review by pitrou — APPROVED** [review]

**Pull request closed without merging**